app/models/conversion_host.rb

```diff
@@ -21,31 +26,97 @@ class ConversionHost < ApplicationRecord
+  # with the conversion host. Otherwise, use the default check which uses the
+  # associated resource's authentications.
+  #
+  # Subclasses should pass provider-specific +options+, such as proxy information.
+  #
+  # This method is necessary to comply with AuthenticationMixin interface.
+  #--
+  # TODO: Use the verify_credentials_ssh method in host.rb? Move that to the
+  # AuthenticationMixin?
+  #
+  def verify_credentials(_auth_type = nil, options = {})
+    if authentications.empty?
+      check_ssh_connection
+    else
+      require 'net/ssh'
+      host = hostname || ipaddress
+
+      authentications.each do |auth|
+        user = auth.userid || ENV['USER'] || Etc.getlogin
+
+        ssh_options = { :timeout => 10, :logger => $log, :verbose => :error }
+
+        if auth.password
+          ssh_options[:password] = auth.password
+          ssh_options[:auth_methods] = %w[password]
+        end
+
+        if auth.auth_key
+          ssh_options[:keys] = [auth.auth_key]
+          ssh_options[:auth_methods] = %w[public_key host_based]
+        end
+
+        # Options from STI subclasses will override the defaults we've set above.
+        ssh_options.merge!(options)
+
+        Net::SSH.start(host, user, ssh_options) { |ssh| ssh.exec!('uname -a') }
+      end
+    end
+  rescue Net::SSH::AuthenticationFailed => err
+    raise MiqException::MiqInvalidCredentialsError, _("Incorrect credentials - %{error_message}") % {:error_message => err.message}
+  rescue Net::SSH::HostKeyMismatch => err
+    raise MiqException::MiqSshUtilHostKeyMismatch, _("Host key mismatch - %{error_message}") % {:error_message => err.message}
+  rescue StandardError => err
+    raise _("Unknown error - %{error_message}") % {:error_message => err.message}
+  else
+    true
+  end
+
+  # Returns a boolean indicating whether or not the conversion host is eligible
+  # for use. To be eligible, a conversion host must have the following properties:
+  #
+  #  - A transport mechanism is configured for source (set by 3rd party).
+  #  - Credentials are set on the conversion host and the SSH connection works.
+  #  - The number of concurrent tasks has not reached the limit.
+  #
   def eligible?
-    source_transport_method.present? && check_ssh_connection && check_concurrent_tasks
+    source_transport_method.present? && verify_credentials && check_concurrent_tasks
   end
 
+  # Returns a boolean indicating whether or not the current number of active tasks
+  # exceeds the maximum number of allowable concurrent tasks specified in settings.
+  #
   def check_concurrent_tasks
     max_tasks = max_concurrent_tasks || Settings.transformation.limits.max_concurrent_tasks_per_host
     active_tasks.size < max_tasks
   end
 
+  # Check to see if we can connect to the conversion host using a simple 'uname -a'
+  # command on the connection. The exact nature of the connection will depend on the
+  # underlying provider.
+  #
   def check_ssh_connection
```

Comment:
Now is the point of this method to see if we can ssh without username/password aka someone set the appliance's root ssh public key to the conversion host's authorized_keys ?

Reply:
Near as I could tell the purpose was just to check that it could make a connection, whether it be via authkey or userid/password or whatever.

Reply:
I wonder if we can delete it then?  @fdupont-redhat do you remember what this was for?

Reply:
The idea was to validate the connection by running a command and not stopping at negotiation... If this validation is provided by AuthenticationMixin, we can remove it.

Reply:
This method would still be used in cases where there wasn't an associated authentication, so we still need it for now. In that case it would default to the resource authentication (redhat) or the provider (openstack).

Reply:
Ohh I see connect_ssh is actually just username/password from the related resource, I thought that was the MiqSshUtil method and was relying on passwordless ssh